src/mem/vmem/vmem_device_memory.c

```diff
@@ -83,7 +85,16 @@ static int munmap_anon(void *addr, size_t size) {
 	 * actual number of pages is (size + 1)/MMU_PAGE_SIZE */
 	int pages = (addr + size - start + MMU_PAGE_SIZE - 1) / MMU_PAGE_SIZE;
 
+	phy_page = mmap_find_phy_page(emmap, start);
+	if (NULL == phy_page) {
+		return SET_ERRNO(ENOENT);
+	}
+
 	phymem_free(start, pages);
```

Comment:
There is a memory leak if pages != phy_page->page_number. Could you check this and in inequal case return EINVAL? That's would be enough for merging this.

But in theory, munmap could be called on range different than was mmaped before.
Consider the following 
```
p1 = mmap(anon, 0x4000) // returns 0x1000
p2 = mmap(anon, 0x4000) // returns 0x6000
```
in result, there is 2 maps 0x1000 - 0x5000 and 0x6000 - 0xa000
As far as I understand munmap interface, munmap(0x0, 0xa000) should munmap both maps.
munmap(0x2000, 0x1000) should result in 3 maps: 0x1000 - 0x2000, 0x3000 - 0x5000 and 0x6000 - 0xa000 (last one unchanged).

I think we could iterate over mmap->page_list and check all kind of intersections of munmapping range and mmaped ranges.
Then we could just alter `page` or `page_number` fileds of phy_page if one of it's end is in unmapping range.
Munmapping range could split one phy_page in two if none of it's ends in munmapping range. One phy_page could be just changed plus new one should be created. Inserting in mmap->page_list is probably OK but potentionally dangerous, since we are modifying list we are iterting over.

So correct implementation is a bit complex, so I suppose merge this PR first, then consider fixing.